README.md

```diff
@@ -5,50 +5,65 @@ Example Heroku add-on built with Rails for the Add-on Partner API v3.
 Following instructions found at
 https://devcenter.heroku.com/articles/building-an-add-on.
 
-## Set up
+See [CONTRIBUTING.md](CONTRIBUTING.md) for information on how to set up and run
+the code for this application.
 
-### macOS
+## Add-on manifest
 
-Install [Homebrew].
+### Heroku docs
 
-Install [Heroku CLI].
+* https://devcenter.heroku.com/articles/building-an-add-on#step-1-generate-your-add-on-manifest
 
-```
-brew install heroku/brew/heroku
-```
+### Related code
 
-Install and start PostgreSQL.
+The manifest is not checked into version control because it contains secrets. A
+copy of the manifest with fake keys is located at [addon-manifest-example.json](addon-manifest-example.json).
 
-```
-brew install postgresql
-brew services start postgresql
-```
+## Basic auth
 
-[Homebrew]: https://brew.sh/
-[Heroku CLI]: https://devcenter.heroku.com/articles/heroku-cli
+### Heroku docs
 
-### Ruby on Rails
+* https://devcenter.heroku.com/articles/building-an-add-on#basic-authentication
 
-This application is built using [Ruby on Rails].
+### Related code
 
-Your system will require [Ruby] to develop on the application.
+Basic auth is implemented via the [`HttpBasicAuth`
+concern](app/controllers/concerns/http_basic_auth.rb). This concern is included
+in [`ApplicationController`](app/controllers/application_controller.rb), which
+all other controllers inherit from, because the Add-on Partner API sends basic
+auth credentials with all requests. Basic auth is selectively skipped for the
+single sign-on views, which are accessed by Heroku customers who are using the
+add-on rather than the Add-on Partner API.
 
-The required Ruby version is listed in the [.ruby-version](.ruby-version) file.
+ The username and password for basic auth are being accessed via environment variables whose values match the `slug` and
+`password` fields in the add-on manifest.
 
-If you do not have this binary, [use this guide to get set up on MacOS].
+## The provisioning request
 
-[Ruby on Rails]: http://rubyonrails.org
-[Ruby]: https://www.ruby-lang.org/en/
-[use this guide to get set up on MacOS]: http://installfest.railsbridge.org/installfest/macintosh
+### Heroku docs
 
-### Configuring the Application
+* https://devcenter.heroku.com/articles/building-an-add-on#the-provisioning-request
+* https://devcenter.heroku.com/articles/getting-started-with-asynchronous-provisioning#implementation
+* https://devcenter.heroku.com/articles/add-on-partner-api-reference#add-on-action-create-provision
+* https://devcenter.heroku.com/articles/platform-api-reference#add-on-action-provision
 
-After cloning this repo, run: `bin/setup`.
+### Related code
 
-## Day-to-day Development
+The `base_url` key in the add-on manifest specifies a path of
+`/heroku/resources`, which is routed to the
+[`Heroku::ResourcesController`](app/controllers/heroku/resources_controller.rb)
+`#create` method in this codebase. The controller looks for the `plan` param,
```

Comment:
maybe say something to the effect of "this somewhat artificial plan slug distinction is just to demonstrate the process of provisioning plans asynchronously and synchronously".

Reply:
my fear is folks will think there's something magic about plan slugs when it related to async / sync provisioning.